evalml/utils/nullable_type_utils.py

```diff
@@ -57,16 +58,31 @@ def _downcast_nullable_y(y, handle_boolean_nullable=True, handle_integer_nullabl
         y with any incompatible nullable types downcasted to compatible equivalents.
     """
     if y.ww.schema is None:
-        y = ww.init_series(y)
+        if isinstance(y, pd.DataFrame):
+            y.ww.init()
+        else:
+            y = ww.init_series(y)
 
     incompatible_logical_types = _get_incompatible_nullable_types(
         handle_boolean_nullable,
         handle_integer_nullable,
     )
 
-    if isinstance(y.ww.logical_type, tuple(incompatible_logical_types)):
-        new_ltype = _determine_downcast_type(y)
-        return y.ww.set_logical_type(new_ltype)
+    if isinstance(y, pd.DataFrame):
```

Comment:
At first glance, this code looks identical to `_downcast_nullable_X` here. Any reason we can't just call that function in this case?